SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +276,37 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession) {
+            NSError *sendErr = nil;
+            
+            if (![strongSelf sdl_dequeueAndWriteToOutputStream:&sendErr] && sendErr != nil) {
+                [strongSelf sdl_handleOutputStreamWriteError:sendErr];
+            }
+        }
+    };
+}
+
+#pragma mark - Lifecycle Destruction
+
+- (void)dealloc {
+    [self.sendDataQueue flush];
+    self.sendDataQueue = nil;
+    self.delegate = nil;
+    self.accessory = nil;
+    self.protocol = nil;
+    self.streamDelegate = nil;
+    self.easession = nil;
+    self.ioStreamThread =  nil;
+    self.canceledSema = nil;
```

Comment:
From Stack Overflow: http://stackoverflow.com/questions/19454189/do-i-ever-have-to-call-dispatch-release-under-arc
All GCD objects are managed by ARC, so you don't need to explicitly manage their memory. What is true for dispatch_queue_t also applies to all the other GCD objects.